tests/workloads/imagenet_resnet/compare.py

```diff
@@ -0,0 +1,131 @@
+import os
+
+# Disable GPU access for both jax and pytorch.
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
+
+from flax import jax_utils
+import jax
+import numpy as np
+import torch
+
+from algorithmic_efficiency import spec
+from algorithmic_efficiency.workloads.imagenet_resnet.imagenet_jax.workload import \
+    ImagenetResNetWorkload as JaxWorkload
+from algorithmic_efficiency.workloads.imagenet_resnet.imagenet_pytorch.workload import \
+    ImagenetResNetWorkload as PytWorkload
+from tests import torch2jax_utils as utils
+
+
+def key_transform(k):
+  new_key = []
+  bn = False
+  for i in k:
+    bn = bn or "BatchNorm" in i
+    if 'ModuleList' in i:
+      continue
+    if 'Linear' in i:
+      if 'NonDynamicallyQuantizableLinear' in i:
+        i = 'out'
+      else:
+        i = i.replace('Linear', 'Dense')
+    elif 'Conv2d' in i:
+      i = i.replace('Conv2d', 'Conv')
+    elif 'BatchNorm2d' in i:
+      i = i.replace('BatchNorm2d', 'BatchNorm')
+    elif 'MHSAwithQS' in i:
+      i = i.replace('MHSAwithQS', 'SelfAttention')
+    elif 'weight' in i:
+      if bn:
+        i = i.replace('weight', 'scale')
+      else:
+        i = i.replace('weight', 'kernel')
+    new_key.append(i)
+  return tuple(new_key)
+
+
+def sd_transform(sd):
+  # pylint: disable=locally-disabled, consider-using-generator
+  keys = sorted(sd.keys())
+  c = -1
+  prev = None
+  for k in keys:
+    if 'Bottleneck' in ''.join(k):
+      if prev is None or prev != k[:2]:
+        prev = k[:2]
+        c += 1
+      new_key = (f'BottleneckResNetBlock_{c}',) + k[2:]
+      if 'Sequential' in ''.join(new_key):
+        new_key = tuple([
+            (i.replace('_0', '_proj') if 'BatchNorm' in i or 'Conv' in i else i)
+            for i in new_key
+            if 'Sequential' not in i
+        ])
+      sd[new_key] = sd[k]
+      del sd[k]
+    elif 'BatchNorm' in k[0] or 'Conv' in k[0]:
+      new_key = (k[0].replace('_0', '_init'), *k[1:])
+      sd[new_key] = sd[k]
+      del sd[k]
+
+  return sd
+
+
+def value_transform(k, value, jax_value):
+  k_str = ''.join(k).lower()
+  if 'conv' in k_str and 'kernel' in k_str:
+    rank = len(value.shape)
+    if rank == 3:
+      return value.permute(2, 1, 0)
+    elif rank == 4:
+      return value.permute(2, 3, 1, 0)
+  elif ('dense' in k_str and 'kernel' in k_str) or ('lstm' in k_str and
+                                                    'kernel' in k_str):
+    return value.t()
+  return value
+
+
+if __name__ == "__main__":
```

Comment:
this block of code seems mostly duplicated across compare.py files. could you abstract it into a test function that returns tuple(bool on whether or not the models match, error info describing diffs if not)?

the inputs could be:
- jaxworkload
- pytorchworkload
- jax model_fn kwargs (a dict with keys 'params, 'augmented_and_preprocessed_input_batch, 'model_state, 'mode, 'rng, 'update_batch_norm')
- pytorch model_fn kwargs
- key_transform
- sd_transform

the reason I suggest this refactor is so that then it will be easier to turn these into unit tests (via pytest or absl test), and so that there is a stricter API for adding support for other workloads

Reply:
Done!